ui/utils.py

```diff
@@ -170,3 +178,126 @@ def write_x509_files():
     """Write the x509 certificate and key to files"""
     write_to_file(settings.MIT_WS_CERTIFICATE_FILE, settings.MIT_WS_CERTIFICATE)
     write_to_file(settings.MIT_WS_PRIVATE_KEY_FILE, settings.MIT_WS_PRIVATE_KEY)
+
+
+def get_video_analytics(video_key):
+    """Get video analytics data from Google Analytics."""
+    ga_client = get_google_analytics_client()
+    ga_response = ga_client.reports().batchGet(
+        body=generate_google_analytics_query(video_key)).execute()
+    try:
+        return parse_google_analytics_response(ga_response)
+    except Exception as exc:
+        raise GoogleAnalyticsException(
+            'Could not parse analytics response') from exc
+
+
+def get_google_analytics_client():
+    """Gets a Google Analytics client.
+
+    Returns:
+        analytics_client: An analytics client
+    """
+    try:
+        if not settings.GA_KEYFILE_JSON:
+            raise RuntimeError("missing 'settings.GA_KEYFILE_JSON'")
+        credentials = ServiceAccountCredentials.from_json_keyfile_dict(
+            json.loads(settings.GA_KEYFILE_JSON))
+        analytics_client = build('analyticsreporting', 'v4',
+                                 credentials=credentials)
+        return analytics_client
+    except Exception as exc:  # pylint: disable=broad-except
+        raise GoogleAnalyticsException('Something went wrong with creating a'
+                                       'GoogleAnaltics client') from exc
+
+
+def generate_google_analytics_query(video_key):
+    """Generates a Google Analytics query.
+
+    Returns:
+        analytics_query: a query dict suitable to use as the body of an
+        analytics 'batchGet' request.
+    """
+    if not settings.GA_VIEW_ID:
+        raise RuntimeError("Missing 'settings.GA_VIEW_ID'.")
+    ga_view_id = settings.GA_VIEW_ID
+    # https://developers.google.com/analytics/devguides/reporting/core/v3/reference
+    START_DATE = '2005-01-01'
+    END_DATE = '9999-01-01'
+    query = {
+        'reportRequests': [
+            {
+                'viewId': ga_view_id,
+                'dateRanges': [{'startDate': START_DATE, 'endDate': END_DATE}],
+                'metrics': [{'expression': 'ga:totalEvents'}],
+                'dimensions': [
+                    # [adorsk, 2018-03]
+                    # Achtung! a high degree of implicit coupling to
+                    # dimension names that have been manually set in GA.
+                    # Hard-coding for now.
+                    {'name': 'ga:eventAction'},
+                    {'name': 'ga:dimension1'}
```

Comment:
`dimension1` is available as a django setting (`settings.GA_DIMENSION_CAMERA`)

Reply:
Ah, nice catch. Updated here, and in test as well.